app/controllers/SignUpController.scala

```diff
@@ -70,7 +70,13 @@ abstract class SignUpController(val applicationClient: ApplicationClient, cacheC
           )
         },
         data => {
-          val appRoute = ApplicationRoute.withName(data.applicationRoute)
+          val appRouteFromForm = ApplicationRoute.withName(data.applicationRoute)
```

Comment:
Also, why don't we make val appRoute = ApplicationRoute.withName(data.applicationRoute) match { case ... }